File: OpenCommercial.scala
import java.net.*
import java.io.*

object OpenCommercial {
  def main(arg: String) {
    
    val keyboard: BufferedReader
    val inputLine: String
    
    keyboard = new BufferedReader(new InputStreamReader(System.in))
    
    System.out.print("Please give me a company name.")
    System.out.flush()
    
    inputLine = keyboard.readLine()
    
    val url: URL = new URL("http://www." + inputLine + ".com")
	
    val connection: HttpURLConnection = (HttpURLConnection) url.openConnection()
	
    val page_reader: Reader = new BufferedReader(connection)
	
    var i: Integer = 0
    var c: String
    var acc: String[]
    while i < 5 {
	c = page_reader.read()
	acc[4 - 1] = c
	i = i + 1
    }
    for s in acc {
        print(s + "\n")
    }
}

File: OpenCommercial.java
/* OpenCommercial.java */

import java.net.*;
import java.io.*;

/**  A class that provides a main function to read five lines of a commercial
 *   Web page and print them in reverse order, given the name of a company.
 */

class OpenCommercial {

    /** Prompts the user for the name X of a company (a single string), opens
     *  the Web site corresponding to www.X.com, and prints the first five lines
     *  of the Web page in reverse order.
     *  @param arg is not used.
     *  @exception Exception thrown if there are any problems parsing the 
     *             user's input or opening the connection.
     */
    public static void main(String[] arg) throws Exception {

	BufferedReader keyboard;
	String inputLine;

	keyboard = new BufferedReader(new InputStreamReader(System.in));

	System.out.print("Please enter the name of a company (without spaces): ");
	System.out.flush();        /* Make sure the line is printed immediately. */
	inputLine = keyboard.readLine();

	/* Replace this comment with your solution.  */
	HttpURLConnection connection;
	connection = new HttpURLConnection("http://www." + inputLine + ".com");
	
	String content;
	content = connection.getContent();
	
	Reader page_reader;
	page_reader = new Reader(content);
	
	Integer i;
	i = 0;
	String c;
	String[] acc;
	while i<5 {
	    while ((c = page_reader.read()) != "\n") {
	    	acc.append(c);
	    }
	    System.out.print(acc);
	    i++;
	}

    }
}
